File: nom-ui/src/app/recipe/models/ingredient.model.ts
import { NutrientValueModel } from '../../nutrient/models/nutrient-value.model';
import { BaseCommonModel } from '../../common/models/_base-common.model';

export interface IngredientModel extends BaseCommonModel {
    name: string;
    description?: string;
    categoryId?: number;
    fdcId?: string;
    curationStatus?: string; // Backend returns this as a string, not curationStatusId as number
    curationStatusId?: number; // Keep for backward compatibility, but backend doesn't return this
    nutrients: NutrientValueModel[];
    allergens?: number[];
    isActive?: boolean;
    authorId: number;
    createdById?: number;
    userId?: number;
}

File: nom-api/Nom.Orch/Models/Recipe/IngredientModel.cs
// File: nom-api/Nom.Orch/Models/Recipe/IngredientModel.cs
namespace Nom.Orch.Models.Recipe
{
    public class IngredientModel
    {
        public long Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? FdcId { get; set; }
        public string? Description { get; set; }
        public List<NutrientValueModel> Nutrients { get; set; } = new();
    }
}